### PR TITLE
[14.0][IMP] l10n_th_partner, multi-lang support

### PR DESCRIPTION
**Pull request by kittiu**
This PR ensure that following objects will support Thai name.

* res.partner
* res.users
* hr.employee

### CODE REVIEW

**Comment by kittiu** [comment]
Currency it works if we switch language first, and then save changes. The compute method is run to get the full name properly.

But if I use the Lang wizard, and click OK, the compute method do not trigger.

@newtratip can you look into this please?

File: l10n_th_partner/models/res_users.py [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 
-from odoo import api, models
+from odoo import api, fields, models
 
 
 class Users(models.Model):
     _inherit = "res.users"
 
+    first_name = fields.Char(translate=True)
+    last_name = fields.Char(translate=True)
```
Comment (newtratip) [inline l10n_th_partner/models/res_users.py]: Why do you have these fields ? In the core odoo use name is firstname and lastname.

**Comment by newtratip** [comment]
**Bug**
- Employee page
  1. Create employee and setting name in EN and TH languages, for the example name in EN = `Boy` and name in TH = `บอย` (User's language is EN)
  2. Change user's language to TH and see in the employee still be `Boy`. Really should be `บอย`.
- User page
  1. Create user and setting firstname,lastname in EN and TH languages (User's language is EN)
  2. Name not computed in TH language
- Contact page
  1. Create contact and setting firstname,lastname in EN and TH languages (User's language is EN)
  2. Name not computed in TH language
  3. Error if company type = Company

This is just note. Now, I do not have the solution. I will look into this again if I have idea.

File: l10n_th_partner/models/res_users.py (hunk 1 above)
Comment (kittiu) [reply l10n_th_partner/models/res_users.py]: May be I am wrong. Just quickly do it. You can change as appropriate.

**Comment by kittiu** [comment]
@newtratip this PR is ready to use yet?

**Comment by newtratip** [comment]
@kittiu. No, I'm at the end of our rope.

**Comment by kittiu** [comment]
@newtratip

**Comment by newtratip** [comment]
@kittiu, I removed fisrtname and lastname in res.user https://github.com/OCA/l10n-thailand/pull/264/commits/41838e07a426be990dba512aa7b942d9d9aab0df because this is inherited from res.partner.

The system will throw error "column res_users.firstname does not exist" when you add fisrtname
Selection_446

**Comment by newtratip** [comment]
@kittiu, As you have concern about translate name on res.users when I removed firstname and lastname. I tested case as following below (My default language is English)
1. Input fistname = First Name and lastname = Last Name in english language
2. Input firstname = ชื่อ and lastname = นามสกุล in thai language
3. Switch my language to Thai, firstname = ชื่อ and lastname = นามสกุล --> **It's correct**
4. Switch my language to English, firstname = First Name and lastname = Last Name --> **It's correct**
5. Input fistname = ชื่อ 2 and lastname = นามสกุล 2 in res.partner
6. Switch my language to Thai, firstname = ชื่อ 2 and lastname = นามสกุล 2 --> **It's correct**

I think no problem for removed these fields, WDYT ?, My video record as below

video2

**Comment by kittiu** [comment]
Ok then. Thanks for testing.

**Comment by newtratip** [comment]
@kittiu I would like to propose feature use_prefix_suffix on company type, if click this partner name will compute by prefix / suffix else not compute

Selection_814

In some case we don't need compute partner name by prefix / suffix but need group by company type in partner list view only.

I should push my feature on this pr or I should open another pr (I concern that it will confict with this pr)

Thanks in advance.

**Comment by kittiu** [comment]
Please continue here.

**Comment by newtratip** [comment]
คำนำหน้าชื่อกับชื่อต้องติดกัน 
Reference: https://www.facebook.com/202065769932378/posts/422070174598602/ 
Fix in https://github.com/OCA/l10n-thailand/pull/264/commits/5d2dd50d37238359fae0ff712cb1e3b8bfe99270

**Comment by Saran440** [comment]
@newtratip this commit https://github.com/OCA/l10n-thailand/commit/5d2dd50d37238359fae0ff712cb1e3b8bfe99270 move on https://github.com/OCA/l10n-thailand/pull/278

**Review by Saran440 — APPROVED** [review]

**Review by ps-tubtim — APPROVED** [review]

**Pull request merged**